library/core/src/num/dec2flt/mod.rs

```diff
@@ -250,8 +250,10 @@ pub fn dec2flt<F: RawFloat>(s: &str) -> Result<F, ParseFloatError> {
         None => return Err(pfe_invalid()),
     };
     num.negative = negative;
-    if let Some(value) = num.try_fast_path::<F>() {
-        return Ok(value);
+    if cfg!(not(feature = "optimize_for_size")) {
```

Comment:
This feels like maybe it deserves a clippy lint: https://github.com/rust-lang/rust-clippy/issues/12918

Absolutely not necessary to change here, but I personally would write:
```suggestion
    if !cfg!(feature = "optimize_for_size") {
```